src/oss-defog/DefoggerTool.cs

```diff
@@ -130,7 +289,53 @@ private static void Main(string[] args)
 
                         foreach (var finding in defoggerTool.Findings)
                         {
-                            Logger.Info("{0}: {1} -> {2}", finding.Filename, finding.EncodedText, finding.DecodedText);
+                            Logger.Info("[String] {0}: {1} -> {2}", finding.Filename, finding.EncodedText, finding.DecodedText);
+                        }
+
+                        var binaryDir = (string?)defoggerTool.Options["save-found-binaries-to"];
+                        var binaryNumber = 0;
+                        foreach (var binaryFinding in defoggerTool.BinaryFindings)
+                        {
+                            Logger.Info("[Binary] {0}: {1} -> {2}", binaryFinding.Filename, binaryFinding.EncodedText, binaryFinding.Type);
+                            if (binaryDir is string)
+                            {
+                                Directory.CreateDirectory(binaryDir);
+                                var path = Path.Combine(binaryDir, binaryFinding.Filename, $"binary-{binaryNumber}");
+                                Logger.Info("Saving to ", path);
+                                var fs = new FileStream(path,FileMode.OpenOrCreate);
+                                binaryFinding.DecodedBinary.CopyTo(fs);
+                                binaryNumber++;
+                            }
+                        }
+
+                        var archiveDir = (string?)defoggerTool.Options["save-archives-to"];
+                        var archiveNumber = 0;
+                        foreach (var archiveFinding in defoggerTool.ArchiveFindings)
+                        {
+                            Logger.Info("[Archive] {0}: {1} -> {2}", archiveFinding.Filename, archiveFinding.EncodedText, archiveFinding.Type);
+                            if (archiveDir is string)
+                            {
+                                Directory.CreateDirectory(archiveDir);
+                                var path = Path.Combine(archiveDir, archiveFinding.Filename, $"archive-{archiveNumber++}");
+                                Logger.Info("Saving to ", path);
+                                var fs = new FileStream(path, FileMode.OpenOrCreate);
+                                archiveFinding.DecodedArchive.CopyTo(fs);
+                                archiveNumber++;
+                            }
+                        }
+
+                        var blobDir = (string?)defoggerTool.Options["save-blobs-to"];
+                        var blobNumber = 0;
+                        foreach (var blobFinding in defoggerTool.NonTextFindings)
+                        {
+                            Logger.Info("[Blob] {0}: {1}", blobFinding.Filename, blobFinding.EncodedText);
+                            if (blobDir is string)
+                            {
+                                var path = Path.Combine(blobDir, blobFinding.Filename, $"archive-{archiveNumber++}");
```

Comment:
Should this be archiveNumber of blobNumber?

Reply:
Fixed.